packages/client/lib/sync/fullsync.ts

```diff
@@ -19,7 +19,7 @@ export class FullSynchronizer extends Synchronizer {
   public runningBlocks: boolean
 
   private stopSyncing: boolean
-  private vmPromise?: Promise<void>
+  private vmPromise?: Promise<void | number>
```

Comment:
This is a `Promise<number>` right?

Reply:
No, we need to keep `void` along the call chain for now since this would otherwise break `Blockchain` interface compatibility, see discussion with Ryan. We can update on next breaking releases though.